sky/tools/sky_utils.py

```diff
@@ -128,11 +128,7 @@ def create_fat_macos_framework(args, dst, fat_framework, arm64_framework, x64_fr
         get_mac_framework_dylib_path(x64_framework)], framework_dylib)
   _set_framework_permissions(fat_framework)
 
-  # Compute dsym output path, if enabled.
-  framework_dsym = None
-  if args.dsym:
-    framework_dsym = os.path.join(dst, get_framework_name(fat_framework) + '.dSYM')
-
+  framework_dsym = fat_framework + '.dSYM' if args.dsym else None
```

Comment:
Note the name change from `FlutterMacOS.dSYM` to `FlutterMacOS.framework.dSYM` here. In the `zip_xcframework_archive` function, we do the same [disgusting hack](https://github.com/flutter/engine/pull/54458/files#diff-09e320619e9e7e0a91fabf0255b5bb0edb9d4a3dbe7937dfdb2646288ec9f41cR222-R239) for backward-compatibility we did for iOS before the tool side change landed, and we removed the need for the archive altogether (#54492).